ws-client/src/transport.rs

```diff
@@ -144,27 +151,30 @@ pub enum WsConnectError {
 	ParseError(#[source] serde_json::error::Error),
 }
 
-/// Creates a new WebSocket connection based on [`WsConfig`](crate::WsConfig) represented as a Sender and Receiver pair.
-pub async fn websocket_connection(config: WsConfig<'_>) -> Result<(Sender, Receiver), WsHandshakeError> {
+/// Perform checks on the `WsConfig` that doesn't require any heavy operations.
+fn validate_ws_offline_settings(config: &WsConfig<'_>) -> Result<(Host, Mode, String), WsHandshakeError> {
```

Comment:
extracted to a helper function to easily test this without having to do an actual to connection to the host.